cmd/api.go

```diff
@@ -163,41 +163,15 @@ func (o *apiOptions) validate(c *config.Config) error {
 
 	// In case we want to scaffold a resource API we need to do some checks
 	if o.doResource {
-		// Skip the following check for v1 as resources aren't tracked
-		if !c.IsV1() {
-			// Check that resource doesn't exist or flag force was set
-			if !o.force {
-				resourceExists := false
-				for _, r := range c.Resources {
-					if r.Group == o.resource.Group &&
-						r.Version == o.resource.Version &&
-						r.Kind == o.resource.Kind {
-						resourceExists = true
-						break
-					}
-				}
-				if resourceExists {
-					return errors.New("API resource already exists")
-				}
-			}
+		// Check that resource doesn't exist or flag force was set
+		if !o.force && c.HasResource(o.resource.GVK()) {
+			return errors.New("API resource already exists")
 		}
 
-		// The following check is v2 specific as multi-group isn't enabled by default
-		if c.IsV2() {
-			// Check the group is the same for single-group projects
-			if !c.MultiGroup {
-				validGroup := true
-				for _, existingGroup := range c.ResourceGroups() {
-					if !strings.EqualFold(o.resource.Group, existingGroup) {
-						validGroup = false
-						break
-					}
-				}
-				if !validGroup {
-					return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
-						"kubebuilder.io/migration/multi-group.html")
-				}
-			}
+		// Check that the provided group can be added to the project
+		if c.IsV2() && !c.MultiGroup && len(c.Resources) != 0 && !c.HasGroup(o.resource.Group) {
+			return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
+				"kubebuilder.io/migration/multi-group.html")
 		}
 	}
 
```

Comment:
```suggestion
}
```

Reply:
Just to clarifies, the suggestion on this is:


```go
	// In case we want to scaffold a resource API we need to do some checks
	if o.doResource {
                 if !c.IsV1 {
                         // Check that resource doesn't exist or flag force was set
		        if !o.force && c.HasResource(o.resource.GVK()) {
			    return errors.New("API resource already exists")
	 	        }

		        // Check that the provided group can be added to the project
		        if !c.MultiGroup !c.HasGroup(o.resource.Group) {
			    return fmt.Errorf("multiple groups are not allowed by default, to enable multi-group visit %s",
				"kubebuilder.io/migration/multi-group.html")
		        }
                 }
		
	}
```

Reply:
So basically your suggestion is to move the version check to the scaffolder insted of having it in the config model? Okey, I'm fine with that.

Reply:
First of all, the check should be `c.IsV2()` instead of `!c.IsV1()`. While currently they are the same, as there is only v1 and v2, they have different semantics.

Saying `c.IsV2()` means thata this is specific logic of v2 while `!c.IsV1()` means that this is general logic of every version except v1. If we use `!c.IsV1()` we are expecting every future version to behave the same v2 does in this sense (having a force flag, a multi-group flag, etc.).   This is setting expectation on future versions, which is a bad coding pattern.

Additionally, `c.IsV2()` is not needed for the first of these two checks, as `c.Resource(...)` will always return `false` for v1 as it doesn't track any.

The `len(c.Resources) != 0` that you removed is needed. Imagine you do a `kubebuilder init ...` then  a `kubebuilder create api ...`. `!c.MultiGroup && !c.HasGroup(o.resource.Group)` will be true printing the error when it is the first resource so it should be admited.

Does it make sense?